### PR TITLE
[TritonGEN]: Add operations for named barrier signal and wait

### PR DESCRIPTION
**Pull request by etiotto**
The `gen.named_barrier_signal` operation signals the arrival of a thread at the current program point. The operator has 2 integer parameters: a `barrier_id` which identifies the named barrier and a `thread_group_count` which specifies the number of threads participating in the barrier.

Once all  participating threads have signaled their arrival, any other participating thread waiting at the `gen.named_barrier_wait` operation associated with the same `barrier_id` can exit the wait operation.

### CODE REVIEW

File: include/triton/Dialect/TritonGEN/IR/TritonGENOps.td [hunk 1]
```diff
@@ -161,6 +161,41 @@ def TritonGEN_BarrierOp : TritonGEN_Op<"barrier"> {
   let assemblyFormat = "attr-dict";
 }
 
+def TritonGEN_NamedBarrierSignalOp : TritonGEN_Op<"named_barrier_signal">,
+  Arguments<(ins I32:$barrier_id, I32:$thread_group_count)> {
```
Comment (Dewei-Wang-sh) [inline include/triton/Dialect/TritonGEN/IR/TritonGENOps.td:165]: wondering, are these  bound to be constant?

**Review by Dewei-Wang-sh — APPROVED** [review]
lgtm

File: include/triton/Dialect/TritonGEN/IR/TritonGENOps.td (hunk 1 above)
Comment (whitneywhtsang) [reply include/triton/Dialect/TritonGEN/IR/TritonGENOps.td:165]: thinking that it may not be know in compile time...

**Review by whitneywhtsang — COMMENTED** [review]
Please open a IGC issue to get the OCL equivalent functions to the GenISA used, and a Triton issue to replace the GenISA usage to OCL functions when available.

**Review by whitneywhtsang — APPROVED** [review]

File: include/triton/Dialect/TritonGEN/IR/TritonGENOps.td (hunk 1 above)
Comment (etiotto) [reply include/triton/Dialect/TritonGEN/IR/TritonGENOps.td:165]: They aren't required to be constants.

**Comment by etiotto** [comment]
> Please open a IGC issue to get the OCL equivalent functions to the GenISA used, and a Triton issue to replace the GenISA usage to OCL functions when available.

For the record here is the GH issue: https://github.com/intel/intel-xpu-backend-for-triton/issues/848

**Pull request merged**